src/features/slackapi/slackapi-utils.ts

```diff
@@ -141,31 +192,43 @@ Have a great week!
    * @param timeRegistrations all time registrations after yesterday
    * @param previousWorkDays dates and the number of today
    */
-  export const postWeeklyMessage = (
+  export const postWeeklyMessage = async (
     weeklyCombinedData: WeeklyCombinedData[],
     timeRegistrations:TimeRegistrations[],
     previousWorkDays: PreviousWorkdayDates,
-    nonProjectTimes: NonProjectTime[]) => {
-    const { weekStartDate, weekEndDate } = TimeUtilities.lastWeekDateProvider();
-    const { yesterday, today } = previousWorkDays;
-
-    weeklyCombinedData.forEach(user => {
-      const { slackId, personId, expected } = user;
-
-      const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
-      const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
-
-      if (!isAway && !firstDayBack) {
-        try {
-          client.chat.postMessage({
-            channel: slackId,
-            text: constructWeeklySummaryMessage(user, weekStartDate.toISODate(), weekEndDate.toISODate())
-          });
-        } catch (error) {
-          console.error(`Error while posting slack messages to user ${user.name}`);
+    nonProjectTimes: NonProjectTime[]): Promise<WeeklyMessageData[]> => {
+    try {
+      const { weekStartDate, weekEndDate } = TimeUtilities.lastWeekDateProvider();
+      const { yesterday, today } = previousWorkDays;
+
+      let messagesRecord: WeeklyMessageData[] = [];
+
+      const promises = weeklyCombinedData.map(user => {
+        const { slackId, personId, expected } = user;
+
+        const isAway = TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, today, nonProjectTimes);
+        const firstDayBack= TimeUtilities.checkIfUserIsAwayOrIsItFirstDayBack(timeRegistrations, personId, expected, yesterday, nonProjectTimes);
+
```

Comment:
```suggestion
```

Reply:
Clarification? Should firstDayBack be removed with the conditional below it?